include/bluetooth/mesh/light_ctrl_reg.h

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/** @file
+ *  @defgroup bt_mesh_light_ctrl_reg Light Lightness Control Regulator
+ *  @ingroup bt_mesh_light_ctrl
+ *  @{
+ *  @brief Light Lightness Control regulator API
+ */
+
+#ifndef BT_MESH_LIGHT_CTRL_REG_H__
+#define BT_MESH_LIGHT_CTRL_REG_H__
+
+#include <bluetooth/mesh.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** Light Ligthness Control regulator configuration. */
+struct bt_mesh_light_ctrl_reg_cfg {
+	/** Regulator upwards integral coefficient */
+	float kiu;
+	/** Regulator downwards integral coefficient */
+	float kid;
+	/** Regulator upwards proportional coefficient */
+	float kpu;
+	/** Regulator downwards proportional coefficient */
+	float kpd;
+	/** Regulator dead zone (in percent) */
+	float accuracy;
+};
+
+/** Common regulator context */
+struct bt_mesh_light_ctrl_reg {
+	/** Initialize the regulator */
+	void (*init)(struct bt_mesh_light_ctrl_reg *reg);
+	/** Start the regulator */
+	void (*start)(struct bt_mesh_light_ctrl_reg *reg);
+	/** Stop the regulator */
+	void (*stop)(struct bt_mesh_light_ctrl_reg *reg);
+	/** Regulator configuration */
+	struct bt_mesh_light_ctrl_reg_cfg cfg;
+	/** Measured value */
+	float measured;
```

Comment:
This name is a bit confusing. Maybe change to original name (`ambient_lux`)? Or just `ambient`.

Reply:
The idea is that the regulator should be generic, and not usable only for lightness but for any other measured value. But as we won't extract it out of the model now, I can change it back for now.

Reply:
Yeah, but it is not clear what does 'measured' here mean: measured regulator output or measured ambient level? Since target value and ambient level are inputs for the regulator, they could be both obtained from light ctrl srv at the same time, by e.g. by modifying `_target_get` function to fetch both values.

Reply:
True, but in the server, they change at different points in time. The target/setpoint changes when requested or on state changes in the state machine, the ambient changes whenever sensor messages arrive. That's why I didn't combine them, to avoid having to set the target every time the ambient light is reported from the sensor.

Reply:
I think it doesn't matter when they are changed if in the regulator they both are only used together.

Reply:
If I keep the interpolation in the regulator, then combining them will complicate things. As I'm leaning towards moving this regulator out of subsys/mesh, the interpolation will probably stay in the regulator.